File: Android/app/src/main/java/com/diveapp/android/repository/UserRepository.kt
package com.diveapp.android.repository

import com.diveapp.android.core.network.apiCall
import com.diveapp.android.model.CurrentUser
import com.diveapp.android.model.ProfileSetupRequest
import com.diveapp.android.model.UserProfileResponse
import com.diveapp.android.service.UserService

class UserRepository(private val userService: UserService) {
    suspend fun fetchCurrentUser(): CurrentUser = apiCall { userService.fetchCurrentUser() }

    suspend fun setupProfile(nickname: String, profileImageUrl: String?, phoneNumber: String?): UserProfileResponse =
        apiCall { userService.setupProfile(ProfileSetupRequest(nickname, profileImageUrl, phoneNumber)) }
}

File: Frontend/DiveApp/Repositories/UserRepository.swift
import Foundation

final class UserRepository {
    private let userService: UserService

    init(userService: UserService) {
        self.userService = userService
    }

    func fetchCurrentUser() async throws -> CurrentUser {
        try await userService.fetchCurrentUser()
    }

    func setupProfile(nickname: String, profileImageURL: String?, phoneNumber: String?) async throws -> UserProfileResponse {
        try await userService.setupProfile(nickname: nickname, profileImageURL: profileImageURL, phoneNumber: phoneNumber)
    }
}
